dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/NotificationRuleActionImplementer.java

```diff
@@ -79,16 +77,7 @@ protected ExternalNotificationLogEntry createLogEntry(String key, String templat
 
   @Transactional(readOnly = true)
   public ProgramNotificationTemplate getNotificationTemplate(RuleAction action) {
-    String uid = "";
-
-    if (action instanceof RuleActionSendMessage) {
-      RuleActionSendMessage sendMessage = (RuleActionSendMessage) action;
-      uid = sendMessage.notification();
-    } else if (action instanceof RuleActionScheduleMessage) {
-      RuleActionScheduleMessage scheduleMessage = (RuleActionScheduleMessage) action;
-      uid = scheduleMessage.notification();
-    }
-
+    String uid = action.getValues().get("notification");
```

Comment:
Wouldn't be possible to use the variable defined in `DefaultProgramRuleEntityMapperService`? It would make it easier to know where it comes from.